### ISSUE TITLE
[Request] Compute HW Guide/Benchmark

### ISSUE THREAD

**Issue by stbdang** [huggingface/lerobot#3191]
### Ticket Type

💡 Feature Request / Improvement

### Environment & System Info

```Shell
LeRobot 0.5.0
MacBook Pro M1 MAX 32GB (32C)
```

### Description

Hi, 

Just started working with LeRobot + SO101, currently running it on M1 Max MacBook Pro (32GB). Training ACT is going to take >13 hours (tried num_workers=2,4 and it didn't make a difference) - probably means I would need to get compute HW/use cloud if I'm planning to train anything. 

Is there a suggested HW list or expected performance benchmarks that I can refer to? This would help people tremendously in making purchase decisions, and people can contribute with their data points if we have a standardised metric (e.g., Training ACT). Any quantitative number would be better than "you should be good with RTX 4080".

### Relevant logs or stack trace

```Shell

```

### Checklist

- [ ] I have searched existing tickets to ensure this isn't a duplicate.
- [x] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.

**Comment by fracapuano** [comment]
Ciao @stbdang 👋 Great to know you're getting started with lerobot and got an SO101 :) To be entirely honest, I think you should be pretty well setup with the amount of compute you mention (ie, a `M1 Max MacBook Pro`. FYI, we benchmarked _test-time_ performance of the most common policies supported by lerobot in our ICLR paper ➡️ https://arxiv.org/abs/2602.22818v1).

I think your question is a bit deeper tho. A general rule of thumb when training any (robot) learning algorithms is that better compute buys you more memory*, which in turn buys you faster training.

That is because you can allow for larger batch-sizes which, given the same model, might make your training anything between 2x-10x faster (depending on your bottleneck). Of course, the main predictor of how long your training is going to be is the _number of training steps_ you've set out to train your model for, `training_steps = n_epochs * (n_samples_in_dataset // batch_size)`. 
I am afraid the number of workers you use can only marginally help here, as it basically helps with speeding up the way batches from the dataset are fed to the model, while your bottleneck are forward/backward passes. 

With all that being said: 13 hours seems a bit excessive for most commonly sized datasets (50-100 demonstrations, which should take a lot less), but it depends on your specific training setup. Something else worth checking is whether you are training with Apple silicon acceleration---`torch.device("mps")` does wonders, and can really speed up training (and inference), compared to CPU-only training. Also, NVIDIA GPUs (`torch.device("cuda")`) can also prove very useful in further accelerating your training, and accommodating larger batch sizes.

**Question for you**
Any change you can share your training setup (ie., how you are running training)?

---
*For the sake of completeness: this is not necessarily true. Sometimes better compute means more CUDA cores with the same vRAM. In that case, better compute buys you faster computations.

**Comment by stbdang** [comment]
@fracapuano Thank you for getting back to me! I took a look at the paper, and I see that MPS inference performance - which I verified on my setup to be usable with ACT. Regarding training, I am using MPS, and the activity shows ~90-95% GPU utilization, indicating that it is indeed using MPS acceleration. I ran the training without any modifications to the steps (believe the default is 100K)/batch_size, and my dataset contained only 35 episodes. I haven't touched the batch size yet, so I'll try that next time (it was using about ~7GB of memory, so I think I can safely double it)

Coming from someone who's just getting started with LeRobot, I love how SO-101+LeRobot is democratizing robotics HW+SW. Just wish there was a feasible path to NVIDIA GPU alternatives that are comparably optimized for PyTorch - seems this is now the long pole in terms of cost. 

Anyway, I'm actually shopping for a used 3090, so hopefully I can do more with this.

**Comment by fracapuano** [comment]
no probs, and congrats on "asking the right questions". just one thing to be careful is that apple silicon has shared memory, so if you're doing something else on your mac while training you might be slowed (slowing) down (by) training.

I'd recommend giving a look at the self-contained tutorial I wrote a while back: https://arxiv.org/abs/2510.12403
It can really help you understand the algorithms, and I would love to know what you think of it!

**Comment by imstevenpmwork** [comment]
Hello! I just created a minimal -and mostly intuitive- guide for answering the most common questions around Hw compute requirements & resources for training a policy in lerobot.

Docs: https://moon-ci-docs.huggingface.co/docs/lerobot/pr_3534/en/hardware_guide
PR: https://github.com/huggingface/lerobot/pull/3534

Feel free to re-open this issue or comment directly in the above PR if you spot anything worth mentioning

Thanks for the contribution and idea 😄
